File: Domain/Application.cs
using System;

namespace Domain
{
    public class Application
    {
        public string AppUserId { get; set; }
        public AppUser AppUser { get; set; }
        public Guid PropertyId { get; set; }
        public Property Property { get; set; }
        public Guid ReferenceNumber { get; set; }
        public DateTime CreationDate { get; set; }
        public string Status { get; set; }
    }
}

File: client-app/src/models/application.ts
import { Property } from "./property";
import { Applicant, ApplicantFormValues } from "./user";

export interface Application {
  appUserId: string;
  propertyId: string;
  applicant: Applicant;
  applicantName: string;
  propertyAddress: string;
  referenceNumber: string;
  property: Property;
  creationDate: Date | null;
  status: string;
}

export class ApplicationFormValues {
  appUserId: string = "";
  propertyId: string = "";
  applicant: ApplicantFormValues = new ApplicantFormValues();
  referenceNumber: string = "";
  property: Property = new Property();
  creationDate: Date | null = null;
  status: string = "";

  constructor(application?: ApplicationFormValues) {
    if (application) {
      this.appUserId = application.appUserId;
      this.propertyId = application.propertyId;
      this.applicant = new ApplicantFormValues();
      this.referenceNumber = application.referenceNumber;
      this.property = application.property;
      this.creationDate = application.creationDate;
      this.status = application.status;
    }
  }
}
